packer/builder/azure/arm/azure_client.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Open Technologies, Inc.
+// All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.
+// See License.txt in the project root for license information.
+
+package arm
+
+import (
+	"github.com/Azure/azure-sdk-for-go/Godeps/_workspace/src/github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/Azure/azure-sdk-for-go/arm/resources"
+	armStorage "github.com/Azure/azure-sdk-for-go/arm/storage"
+	"github.com/Azure/azure-sdk-for-go/storage"
+)
+
+type AzureClient struct {
+	compute.VirtualMachinesClient
+	network.PublicIPAddressesClient
+	resources.GroupsClient
+	resources.DeploymentsClient
+	storage.BlobStorageClient
+}
+
+func NewAzureClient(subscriptionID string, resourceGroupName string, storageAccountName string, servicePrincipalToken *azure.ServicePrincipalToken) (*AzureClient, error) {
+	var azureClient = &AzureClient{}
+
+	azureClient.DeploymentsClient = resources.NewDeploymentsClient(subscriptionID)
+	azureClient.DeploymentsClient.Authorizer = servicePrincipalToken
+
+	azureClient.GroupsClient = resources.NewGroupsClient(subscriptionID)
+	azureClient.GroupsClient.Authorizer = servicePrincipalToken
+
+	azureClient.PublicIPAddressesClient = network.NewPublicIPAddressesClient(subscriptionID)
+	azureClient.PublicIPAddressesClient.Authorizer = servicePrincipalToken
+
+	azureClient.VirtualMachinesClient = compute.NewVirtualMachinesClient(subscriptionID)
+	azureClient.VirtualMachinesClient.Authorizer = servicePrincipalToken
+
+	accountsClient := armStorage.NewAccountsClient(subscriptionID)
```

Comment:
nit: maybe use 'storageAccounts' vs. 'accounts' to indicate the type of account?